src/webgpu/api/validation/texture/bgra8unorm_storage.spec.ts

```diff
@@ -108,3 +111,95 @@ if the feature bgra8unorm-storage is not enabled.
 
     t.testCreateShaderModuleWithBGRA8UnormStorage(shaderType, false);
   });
+
+g.test('configure_storage_usage_on_canvas_context_without_bgra8unorm_storage')
+  .desc(
+    `
+Test that it is invalid to configure a GPUCanvasContext with GPUStorageBinding usage and a GPUDevice
+without 'bgra8unorm-storage' enabled.
+`
+  )
+  .params(u =>
+    u
+      .combine('canvasType', kAllCanvasTypes)
+      .beginSubcases()
+      .expand('usage', p => {
+        const usageSet = new Set<number>();
+        for (const usage0 of kTextureUsages) {
+          for (const usage1 of kTextureUsages) {
+            usageSet.add(usage0 | usage1);
+          }
+        }
+        return usageSet;
+      })
+  )
+  .fn(t => {
+    const { canvasType, usage } = t.params;
+    const canvas = createCanvas(t, canvasType, 1, 1);
+    const ctx = canvas.getContext('webgpu');
+    assert(ctx instanceof GPUCanvasContext, 'Failed to get WebGPU context from canvas');
+
+    const incompatible = usage & GPUTextureUsage.STORAGE_BINDING;
```

Comment:
How about `const requiredStorageBinding = !!(usage & GPUTextureUsage.STORAGE_BINDING)`, and